sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseDataCollector.java

```diff
@@ -413,4 +451,94 @@ static CountAndDuration decodeCountAndDuration(long countAndDuration) {
             return new CountAndDuration(countAndDuration >> 44, countAndDuration & MAX_DURATION);
         }
     }
+
+    class OpenTelMetricsStorage {
+        private volatile ConcurrentHashMap<String, OpenTelMetric> metrics = new ConcurrentHashMap<>();
+        // setting most amount of openTel metrics a user can stream to this. will review if users surpass threshold.
+        private final int maxMetricsLimit = 50;
+        // setting buffer time for receiving data for a metric before declaring it inactive. (in minutes)
+        private final double metricBufferTime = 5;
+
+        public void addMetric(String metricName, double value) {
+            OpenTelMetric metric = metrics.get(metricName);
+            if (metric == null) {
+                if (metrics.size() <= maxMetricsLimit) {
+                    // create new metric and add to metrics map
+                    // (if we have reached the limit, we will not create a new metric and add it to the map
+                    metric = new OpenTelMetric(metricName);
+                    metric.addDataPoint(value);
+                    metrics.putIfAbsent(metricName, metric);
+                }
+            } else {
+                metric.addDataPoint(value);
+            }
+        }
+
+        public ArrayList<QuickPulseMetrics> processMetrics() {
+            ConcurrentHashMap<String, ArrayList<QuickPulseConfiguration.OpenTelMetricInfo>> requestedMetrics = quickPulseConfiguration.getMetrics();
+            ArrayList<QuickPulseMetrics> processedMetrics = new ArrayList<>();
+
+            Iterator<Map.Entry<String, OpenTelMetric>> iterator = this.metrics.entrySet().iterator();
+            while (iterator.hasNext()) {
+                Map.Entry<String, OpenTelMetric> entry = iterator.next();
+                String key = entry.getKey();
+                OpenTelMetric value = entry.getValue();
+
+                if (requestedMetrics.containsKey(key)) {
+                    for (QuickPulseConfiguration.OpenTelMetricInfo metricInfo : requestedMetrics.get(key)) {
+                        QuickPulseMetrics processedMetric = processMetric(value, metricInfo);
+                        processedMetrics.add(processedMetric);
+                    }
+
+                }
+
+                if (ChronoUnit.MINUTES.between(value.getLastTimestamp(), LocalDateTime.now()) > metricBufferTime) {
```

Comment:
nit: I'm not sure if we use local time vs UTC time in the timestamps that are associated with these metrics. It is worth checking which kind of timestamps are in use here & if this block is doing the right comparison.

Reply:
Another note - considering that processMetrics gets called on every post request (1 second), do you envision a scenario where the last timestamp of a particular otel metric received from javaagent is over five minutes ago? Trying to understand in which circumstances would this statement actually be relevant?

Reply:
got rid of the timestamping mechanism.